funs/data_to_tool_fun.R

```diff
@@ -89,3 +89,102 @@ add_uncertainty_sensor <- function(data_measurements, uc_pm10 = 8.5, uc_pm25 = 5
                                         T ~ 0))
   return(data_measurements)
 }
+
+add_uncertainty_sensor_percent <- function(data_measurements, uc_pm10 = 20,
+                                           uc_pm25 = 20, uc_min_pm10 = 8.5,
+                                           uc_min_pm25 = 5.3){
+  # Function to add a percentage of the value as uncertainty to the sensor data, with
+  # a minimal uncertainty. Only valid for sensors, the official stations get sd of 0
+  # arguments:
+  #   data_measurements: dataframe
+  #   uc_pm10: double, to indicate the percentage uncertainty in pm10, default = 20
+  #   uc_pm25: double, to indicate the percentage uncertainty in pm25, default = 20
+  #   uc_pm10: double, to indicate the minimal uncertainty in pm10, default = 8.5
+  #   uc_pm25: double, to indicate the minimal uncertainty in pm25, default = 5.3
+  # return: dataframe as data_measurements with extra column "sd",
+  # this column is 0 for reference stations
+
+  # Add uncertainty to the measurements of the sensors
+  data_measurements <- data_measurements %>%
+    dplyr::mutate(sd = dplyr::case_when(parameter == "pm25_kal" & !grepl("NL", station) ~ (uc_pm25/100*value),
```

Comment:
This assumes that the string "NL" never occurs in the name of a sensor/station. I'm not sure about this. Is it possible to add a more restrictive regex, like "^NL.*[0-9].*" (from the top of my head, use at your own risk ;-)

Reply:
See commit